emukit/quadrature/methods/warpings.py

```diff
@@ -1,30 +1,34 @@
+import abc
 import numpy as np
 from typing import Optional
 
 
-class Warping:
+class Warping(abc.ABC):
     """The warping for warped Bayesian quadrature."""
 
+    @abc.abstractmethod
     def transform(self, Y: np.ndarray) -> np.ndarray:
         """Transform from base-GP to integrand.
 
         :param Y: Function values of latent function, shape (n_points, 1).
         :return: Transformed values, shape (n_points, 1).
         """
-        raise NotImplemented
+        pass
```

Comment:
It seems invalid to have a warping that doesn't do or return anything on transform. I'd say this abstract method should keep throwing an exception

Reply:
well, yeah. It only is relevant really if someone feels like calling the method of the abstract super class. with `super().transform(Y)`. Then, instead of doing nothing it would throw an error. I can change it if you like.

Reply:
I do. But there is a more explicit problem there, no? If I inherit this class and don't override the method, then someone calls it and is left very perplexed

Reply:
I don't think you can instantiate a child class that does not override all abstract methods (it throws an exception).

Reply:
Oh, does it? If so, it makes no difference indeed. But! If you explicitly throw an exception, none of that dialog would have happened!

So I think the choice here is either keep `pass` and document the what you just replied above, or throw an exception.

Reply:
I am not sure where the confusion comes from to be honest. `pass` is the default to use with `abc.abstractmethod`. It's virtually in any abc tutorial or code you can find, so it should not be confusing in the first place (we even use it in Emukit e.g. [here](https://github.com/EmuKit/emukit/blob/main/emukit/core/acquisition/acquisition.py)). One of the main advantages about the `abc` package is in fact that you can force implementations of methods; this, one should not need to explain either. 

The example I gave where it actually matters what to write into the `abtractmethod` is quite cornercase and I hightly doubt many people know that it may have any effect to write sth other than `pass` into an abstract method.  I'll throw an exception just to be extra cautious, although it;s a bit non standard I suppose.

Reply:
Right, then it's really me being behind on how ABC works. THanks for explanation!